README.md

```diff
@@ -253,30 +252,68 @@ Lastly, create a Makefile in the `code` directory to invoke the harness runner a
+### TL;DR
 
-1. git clone https://github.com/confluentinc/kafka-tutorials.git
-2. cd kafka-tutorials
+To run a tutorial programmatically, do the following steps.
 
-Once you have the kafka-tutorials repo checked out do the following:
+1. Check out the kafka-tutorials GitHub repo:
 
-1. Create a `Dockerfile` with the following content:
-    ```text
-     FROM python:3.7-slim
-     RUN pip3 install pyyaml
-    ```
-2. Then run the following command to build and execute the docker image:
-    * `docker build -t runner . ; docker run -v ${PWD}/harness_runner:/harness_runner/ -it --rm runner bash -c 'cd /harness_runner/ && pip3 install -e .'`
+```bash
+git clone https://github.com/confluentinc/kafka-tutorials.git
+cd kafka-tutorials
+```
+
+2. Install the packages for the harness runner.
+
+If you have `pip3` installed locally:
+
+```bash
+(cd /harness_runner/ && pip3 install -e .)
+```
+
+If you don't have `pip3` installed locally, create a `Dockerfile` with the following content:
+
+```text
+FROM python:3.7-slim
+RUN pip3 install pyyaml
+```
+
+and then run the following command to build and execute the Docker image:
+
+```
+docker build -t runner . ; docker run -v ${PWD}/harness_runner:/harness_runner/ -it --rm runner bash -c 'cd /harness_runner/ && pip3 install -e .'
+```
+
+3. Install [gradle](https://gradle.org/install/) for tutorials that compile any code.
 
-3. run the `make` command
-   * `(cd _includes/tutorials/<tutorial name>/<type>/code && make)` where type is one of `ksql | kstreams | kafka`.
-   * For example `(cd _includes/tutorials/transforming/kstreams/code/ && make)`
+4. End-to-end: run the harness runner for a single tutorial by calling `make`, across all `dev`, `test`, and `prod` stages, to validate it works end-to-end. Identify the tutorial you want and then run `make`. Note that this destroys all the resources and Docker containers it created, so it cleans up after itself.  Format: `(cd _includes/tutorials/<tutorial name>/<type>/code && make)` where type is one of `ksql | kstreams | kafka`. Example:
+
+```
+(cd _includes/tutorials/transforming/kstreams/code/ && make)
+```
+
+5. Run-and-play: run the harness runner for a single tutorial by calling `make SEQUENCE="dev, test"`, just across `dev` and `test` stages, which leaves all resources and Docker containers running so you can then play with it.  Format: `(cd _includes/tutorials/<tutorial name>/<type>/code && make SEQUENCE="dev, test")` where type is one of `ksql | kstreams | kafka`. Example:
+
+```
+(cd _includes/tutorials/transforming/kstreams/code/ && make SEQUENCE="dev, test")
+```
+
+Now you can play with the environment, some sample commands shown below.  Don't forget to shutdown Docker containers when you are done.
+
+```
+docker exec broker kafka-topics --list --bootstrap-server localhost:9092
+docker exec -it ksqldb-cli ksql http://ksqldb-server:8088               
+```
```

Comment:
Now you can play with the environment, some sample commands shown below. 

```
docker exec broker kafka-topics --list --bootstrap-server localhost:9092
docker exec -it ksqldb-cli ksql http://ksqldb-server:8088   
```            
Don't forget to shutdown Docker containers when you are done

```
docker-compose down --volumes 
```

Reply:
@bbejeck `docker-compose ` commands work only when in the same directory as the docker-compose.yml file.  Given that we used paranthesis above, `docker-compose down` won't work.

Even more so, when I go to the `code` directory, `docker-compose down` still doesn't work, even when specifying the file:

```
[13:41:49] ~/git/kafka-tutorials/_includes/tutorials/transforming/kstreams/code(GH-453) ✗: docker-compose -f docker-compose.yml down
Removing network code_default
WARNING: Network code_default not found.
```

What if we just tell users how to get docker container info with:

```
docker container ls
docker container rm <id>
```

Reply:
yep that works for me

Reply:
@bbejeck Unfortunately `docker container ls` does not provide the tutorial name:

```
[14:09:06] ~/git/kafka-tutorials(GH-453) ✗: docker container ls
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS              PORTS                                        NAMES
7e9fc9ffaf24        confluentinc/ksqldb-server:0.7.1         "/usr/bin/docker/run"    12 seconds ago      Up 11 seconds       0.0.0.0:8088->8088/tcp                       ksqldb-server
f9441e503e6a        confluentinc/cp-schema-registry:5.5.0    "/etc/confluent/dock…"   23 minutes ago      Up 23 minutes       0.0.0.0:8081->8081/tcp                       schema-registry
d0b328c59187        confluentinc/cp-enterprise-kafka:5.5.0   "/etc/confluent/dock…"   23 minutes ago      Up 23 minutes       9092/tcp, 0.0.0.0:29092->29092/tcp           broker
3d321fee7116        confluentinc/cp-zookeeper:5.5.0          "/etc/confluent/dock…"   23 minutes ago      Up 23 minutes       2888/tcp, 0.0.0.0:2181->2181/tcp, 3888/tcp   zookeeper
```

Reply:
I meant `docker network ls` since it seems the tutorial name is used for the network name.  
At any rate, what you have is fine